### PR TITLE
Promote e2e verifying different types of services and their behaviours

### PR DESCRIPTION
**Pull request by mgdevstack**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

 /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Promotes following e2e to Conformance as they veify important functional behavior of changing service from one type to another and ability to expose same service port for different protocol for a NodePort type service -

1. ~~[sig-network] Services should be able to update NodePorts with two same port numbers but different protocols~~
[TODO: need to test reachability after service update]
2. [sig-network] Services should be able to change the type from ExternalName to ClusterIP
3. [sig-network] Services should be able to change the type from ExternalName to NodePort
4. [sig-network] Services should be able to change the type from ClusterIP to ExternalName
5. [sig-network] Services should be able to change the type from NodePort to ExternalName
6. ~~[sig-network] Services should use same NodePort with same port but different protocols~~
[TODO: need to test reachability after service update]

**Which issue(s) this PR fixes**:

Fixes #76441

**Special notes for your reviewer**:
- All above e2e has execution time ~6.410 seconds.
- No flakes found on Kind cluster. (Testgrid flake links would be added in comment after few days of observation)

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/area conformance
/sig testing
@kubernetes/sig-node-pr-reviews
@kubernetes/sig-architecture-pr-reviews
@kubernetes/cncf-conformance-wg

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
Adding @kubernetes/sig-network-pr-reviews as the tests fall under their domain :)

**Comment by johnbelamaric** [comment]
/lgtm

As mentioned on the previous accidentally closed PR, these are OK but not great. They check that the changes are made in the API but they aren't really e2e tests in that they don't validate the actual service is working according to the spec. I am OK with merging them as is but we need to create tests that actually check the functionality of each service as it changes type.

**Review by cmluciano — CHANGES_REQUESTED** [review]
I agree with @johnbelamaric 's concerns about some of the tests only validating the API. Other net conformance tests test the implementation & I believe we should hold on merging this PR without them.

**Comment by bclau** [comment]
FWIW, the tests pass on Windows.

**Comment by mgdevstack** [comment]
/assign
To update e2e wrt. https://github.com/kubernetes/kubernetes/pull/77865#issuecomment-492849235 verifying service behaviours not just limited to API updates.

**Comment by johnbelamaric** [comment]
/lgtm cancel

Let's get the data plane checks in before making this conformance, or we will create a false sense of coverage.

**Comment by mgdevstack** [comment]
/hold
untill https://github.com/kubernetes/kubernetes/pull/77865#issuecomment-499347854 is addressed.

**Comment by johnbelamaric** [comment]
Status:
 * Waiting merge of https://github.com/kubernetes/kubernetes/pull/79667
 * Once that merges, we must monitor these for flakiness
 * Assuming no flakiness, this will be ready to promote

**Comment by spiffxp** [comment]
I'm punting this back to In Progress until we're ready to review again

File: test/e2e/network/service.go
Comment (mgdevstack) [inline test/e2e/network/service.go]: Once TODO: is addressed, we can promote in separate PR.

**Comment by mgdevstack** [comment]
These e2e are flaky. Once #81341 is merged and testgrid signals green, this PR would be rebased.

**Comment by mgdevstack** [comment]
Rebased and observing for flakiness.

**Comment by spiffxp** [comment]
>[sig-network] Services should be able to change the type from ExternalName to ClusterIP
>[sig-network] Services should be able to change the type from ExternalName to NodePort
>[sig-network] Services should be able to change the type from ClusterIP to ExternalName
>[sig-network] Services should be able to change the type from NodePort to ExternalName

These are non-flaky and run reasonably fast: https://testgrid.k8s.io/sig-release-master-blocking#gce-cos-master-default&width=5&include-filter-by-regex=change%20the%20type&graph-metrics=test-duration-minutes

These are no longer mere-CRUD tests, but actually exercise the services

/lgtm

/assign @bgrant0607 @smarterclayton 
for conformance /approve

**Comment by smarterclayton** [comment]
/approve

High value, low risk, key guarantees provided by services.

**Pull request merged**
